### PR TITLE
Ensure that `make test-e2e-node` can be executed anywhere 

### PR DESCRIPTION
**Pull request by zouyee**
Signed-off-by: Zou Nengren <zouyee1989@gmail.com>

**What type of PR is this?**
 /kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
a. Ensure that `make test-e2e-node` can be executed anywher
b. using vendor instead

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: hack/make-rules/test-e2e-node.sh [hunk 1]
```diff
@@ -177,7 +177,7 @@ else
 
   # Test using the host the script was run on
   # Provided for backwards compatibility
-  go run test/e2e_node/runner/local/run_local.go \
+  go run --mod=vendor test/e2e_node/runner/local/run_local.go \
```
Comment (liggitt) [inline hack/make-rules/test-e2e-node.sh]: /hold

we are not currently building in module mode (and will likely explicitly disable module mode for builds when we first move to go1.13, see proposed plan https://github.com/kubernetes/kubernetes/issues/82531)

File: test/utils/paths.go [hunk 2]
```diff
@@ -50,8 +50,11 @@ func RootDir() (string, error) {
 	path := filepath.Dir(testExec)
 
 	// Look for the kubernetes source root directory
-	if strings.Contains(path, "k8s.io/kubernetes") {
-		splitPath := strings.Split(path, "k8s.io/kubernetes")
+	if strings.Contains(path, "kubernetes") {
```
Comment (liggitt) [inline test/utils/paths.go]: this function has lots of callers, is this change reasonable for all of them?

Comment (liggitt) [reply test/utils/paths.go]: a single `kubernetes` segment is much less of a signal than the fully-qualified component name. should we look for something more indicative like the name *and* a source file known to be in the root?

Comment (BenTheElder) [reply test/utils/paths.go]: if we're going to look for this we can check for something more substantial like the go.mod specifying that it is the k8s.io/kubernetes module

Comment (zouyee) [reply test/utils/paths.go]: > a single `kubernetes` segment is much less of a signal than the fully-qualified component name. should we look for something more indicative like the name _and_ a source file known to be in the root?

I agree with @BenTheElder  that we can't judge by name alone. Besides, do we lack some documentation about these situations?

**Pull request closed without merging**